src/packages/end-to-end/src/__tests__/api/storage-provider/issues/gh1458.test.ts

```diff
@@ -0,0 +1,144 @@
+import { randomUUID } from 'crypto';
+import fs from 'fs';
+import gql from 'graphql-tag';
+import { Jimp } from 'jimp';
+import path from 'path';
+import request from 'supertest-graphql';
+import { config } from '../../../../config';
+
+type Media = {
+	filename: string;
+	type: string;
+	url: string;
+};
+
+type Submission = {
+	id: string;
+	image: Media;
+};
+
+describe('Nested entities in custom operations', () => {
+	let testSubmission: Submission;
+
+	beforeEach(async () => {
+		const uuid = randomUUID();
+		const filename = `${uuid}.png`;
+		const newUpload = await request<{ getUploadUrl: { url: string; filename: string } }>(
+			config.baseUrl
+		).path('/').query(gql`
+			mutation {
+				getUploadUrl( key: "${filename}" ) 
+			}
+		`);
+		const file = fs.readFileSync(path.join(__dirname, '../fixtures/pickle.png'));
+		const uploadUrl = newUpload.data?.getUploadUrl.url;
+		const uploadedFilename = newUpload.data?.getUploadUrl.filename;
+		if (!uploadUrl) {
+			throw new Error('No URL returned');
+		}
+		if (!uploadedFilename) {
+			throw new Error('No filename returned');
+		}
+		await fetch(uploadUrl, {
+			method: 'PUT',
+			body: file,
+		});
+
+		const newSubmission = await request<{ createSubmission: Submission }>(config.baseUrl).path('/')
+			.query(gql`
+			mutation {
+				createSubmission(input: { image: { filename: "${uploadedFilename}", type: IMAGE } }) {
```

Comment:
I think you can pass this in as a variable to the `.query()` call, no? Would be a bit more maintainable.